### PR TITLE
Making clap an optional feature

### PR DESCRIPTION
**Pull request by elichai**
This saves around 10 dependencies for people who uses `cbindgen` in the build.rs only (so they don't need clap)

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -14,11 +14,10 @@ repository = "https://github.com/eqrion/cbindgen/"
 travis-ci = { repository = "eqrion/cbindgen" }
 
 [dependencies]
-clap = "2"
+clap = { version = "2", optional = true }
 log = "0.4"
-serde = {version = "1.0", features = ["derive"]}
+serde = { version = "1.0", default-features = false, features = ["derive"]}
```
Comment (emilio) [inline Cargo.toml]: Should this be part of this commit? Or should it be a different commit?

Comment (elichai) [reply Cargo.toml]: `serde-derive` is redundant if you use the `derive` feature, and the default features aren't needed here. I can rebase it to a different commit.

Comment (elichai) [reply Cargo.toml]: Rebased.

**Comment by emilio** [comment]
Thanks, looks great :)
